torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -575,6 +575,69 @@ class ScriptModuleSerializer : public ScriptModuleSerializerBase {
   caffe2::serialize::PyTorchStreamWriter writer_concrete_;
 };
 
+void ScriptModuleSerializerUniversal::serialize(
+    Module &module, 
+    const std::string &ts_id) {
+  // Serialize the model object
+  writeArchive("data", module._ivalue(), ts_id);
+  // Then we serialize all code info.
+  convertTypes(module.type());
+  // The tensor constants from the code are written to a separate archive
+  // so loading the code does not depend on loading the data
+  std::vector<IValue> ivalue_constants(constant_table_.begin(), constant_table_.end());
+  writeArchive("constants", c10::ivalue::Tuple::create(ivalue_constants), ts_id);
+}
+
+void ScriptModuleSerializerUniversal::writeArchive(
+    const std::string &archive_name,
+    const IValue &value,
+    const std::string &ts_id) {
+  std::vector<char> data;
+  // Vector to capture the run-time class types during pickling the IValues
+  std::vector<c10::ClassTypePtr> memoizedClassTypes;
+  // from the exporter
+  std::vector<std::string> tensor_names;
+  Pickler data_pickle(
+      [&](const char *buf, size_t size) {
+        data.insert(data.end(), buf, buf + size);
+      },
+      nullptr,
+      [&](const c10::ClassTypePtr &t) {
+        return type_name_uniquer_.getUniqueName(t);
+      },
+      &memoizedClassTypes,
+      [&]() {
+        // returns a string to use in picker.cpp as storage obj key
+        tensor_names.push_back(getNextStorageID() + ".storage");
+        return tensor_names.back();
+      });
+  data_pickle.protocol();
+  data_pickle.pushIValue(value);
+  data_pickle.stop();
+
+  size_t i = 0;
+  assert(tensor_names.size() == data_pickle.tensorData().size());
+  for (const auto &td : data_pickle.tensorData()) {
+    WriteableTensorData writable_td = getWriteableTensorData(td);
+    std::string fname = ".data/" + tensor_names[i++];
+    writer_.writeRecord(fname, writable_td.data(), writable_td.sizeInBytes());
+  }
+
+  std::string fname = ".data/ts_code/" + ts_id + "/" + archive_name + ".pkl";
+  writer_.writeRecord(fname, data.data(), data.size());
+
+  // serialize all the captured run-time class types
+  for (const c10::ClassTypePtr &wroteType : memoizedClassTypes) {
+    convertNamedType(wroteType);
+  }
+}
+
+std::string ScriptModuleSerializerUniversal::getNextStorageID() {
+  // call exporter's get_storage_id function
+  py::object result = package_exporter.attr("get_storage_id")();
```

Comment:
This call right here is what is causing the build errors; I can't figure out what to add to the build_variables.bzl to get it to compile correctly. I get the following errors: 
```
/home/lillianjohnson/local/pytorch/build/lib/libtorch_cpu.so: undefined reference to `PyInstanceMethod_Type'
/home/lillianjohnson/local/pytorch/build/lib/libtorch_cpu.so: undefined reference to `PyDict_DelItemString'
/home/lillianjohnson/local/pytorch/build/lib/libtorch_cpu.so: undefined reference to `_Py_Dealloc'
```